uast/utils.go

```diff
@@ -66,13 +66,72 @@ func CyclomaticComplexity(n *Node)  int {
 		}
 		n := p.Node()
 		for _, r := range n.Roles {
-			switch(r) {
+			switch r {
 			case If, SwitchCase, For, ForEach, While,
-			     DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
-			     OpBooleanXor:
+				DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
+				OpBooleanXor:
 				complx++
 			}
 		}
 	}
 	return complx
 }
+
+func (n *Node) ChildrenOfRole(wanted Role) []*Node {
+	var children []*Node
+	for _, child := range n.Children {
+		for _, role := range child.Roles {
```

Comment:
Better yet, use a visitor for the slice and pass it a clousure for counting, adding to a slice...